snmp/datadog_checks/snmp/config.py

```diff
@@ -297,9 +297,49 @@ def add_uptime_metric(self):
             return
         # Reference sysUpTimeInstance directly, see http://oidref.com/1.3.6.1.2.1.1.3.0
         uptime_oid = OID('1.3.6.1.2.1.1.3.0')
-        self.scalar_oids.append(uptime_oid)
+        self.oid_config.add_parsed_oids([uptime_oid], [], [])
         self._resolver.register(uptime_oid, 'sysUpTimeInstance')
 
         parsed_metric = ParsedSymbolMetric('sysUpTimeInstance', forced_type='gauge')
         self.parsed_metrics.append(parsed_metric)
         self._uptime_metric_added = True
+
+
+class OIDConfig(object):
+    """
+    Manages scalar/next/bulk oids.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._parsed_scalar_oids = []  # type: List[OID]
+        self._parsed_next_oids = []  # type: List[OID]
+        self._parsed_bulk_oids = []  # type: List[OID]
+
+    @property
+    def scalar_oids(self):
+        # type: () -> List[OID]
+        return self._parsed_scalar_oids
+
+    @property
+    def next_oids(self):
+        # type: () -> List[OID]
+        return self._parsed_next_oids
+
+    @property
+    def bulk_oids(self):
+        # type: () -> List[OID]
+        return self._parsed_bulk_oids
+
+    def add_parsed_oids(self, parsed_scalar_oids, parsed_next_oids, parsed_bulk_oids):
+        # type: (List[OID], List[OID], List[OID]) -> None
+        self._parsed_scalar_oids.extend(parsed_scalar_oids)
+        self._parsed_next_oids.extend(parsed_next_oids)
+        self._parsed_bulk_oids.extend(parsed_bulk_oids)
```

Comment:
(Nit) Suggestion to reduce verbosity.

- Ideally the parameters should be kw-only, so `.add(scalar=..., next=..., bulk=...)`. But because of Python 2 we can't enforce it. Still I think we should stick to kw-only usage for readability.
- Adding defaults allows doing `oid_config.add(scalar=[...])` in tests.
- `Sequence` instead of `List` allows using empty tuples as defaults (instead of defaulting to `None`  and then doing `scalar or []`). This is possible because `.extend()` accepts any iterable.

```suggestion
    def add(self, scalar=(), next=(), bulk=()):
        # type: (Sequence[OID], Sequence[OID], Sequence[OID]) -> None
        self._scalar.extend(scalar)
        self._next.extend(next)
        self._bulk.extend(bulk)
```

Reply:
Updated to remove `parsed` term, but kept `_oids` to avoid `next` shadowing the builtin.